docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,120 @@
+your frontend and DAML artefacts of your project to a production ledger. See :ref:`Deploying to DAML
+Ledgers <deploy-ref_index>` for an in depth manual for specific ledgers.
 
-The critical thing here is that each event is processed in isolation - the application does not need to keep any application-related state between each event. It is this that differentiates it from a state-driven application.
+Authentication
+~~~~~~~~~~~~~~
 
-To do this, the application should:
+DAML ledgers support a unified interface for authentication of commands, a detailed description can
+be found :ref:`here <authentication>`. Some DAML ledgers like for example https://projectdabl.com
+offer an integrated authentication service, but you can also use an external service provider for
+authentication like https://auth0.com. The DAML react libraries support interfacing with an
+authenticated DAML ledger. Simply initialize your ``DamlLedger`` object with the token obtained by
+an authentication service. How authentication works and the form of the required tokens is described
+in the :ref:`Authentication <authentication>` section.
 
-1. Create a connection to the Transaction Service, and instantiate a stream handler to handle the new event stream. By default, this will read events from the beginning of the ledger. This is usually not what is wanted, as it may replay already processed transactions. In this case, the application can request the stream from the current ledger end. This will, however, cause any events between the last read point and the current ledger end to be missed. If the application must start reading from the point it last stopped, it must record that point and explicitly restart the event stream from there.
+Developer workflow
+~~~~~~~~~~~~~~~~~~
 
-2. Optionally, create a connection to the Command Submission Service to send any required commands back to the ledger.
+The DAML SDK enables a local development environment with fast iteration cycles. If you run
+``daml-start.sh`` of the ``create-daml-app``, a local DAML sandbox ledger is started that is updated
+with your most recent DAML code on any change. Next, you can start your frontend in development mode
+by changing to your ``ui`` directory and run ``yarn start``. This will reload your frontend whenever
+you make changes to it. You can add unit tests for your DAML models by writing :ref:`DAML scenarios
+<testing-using-scenarios>`. These will also be reevaluated on change.
+A typical DAML developer workflow is to
 
-3. Act on the content of events (type, content) to perform any action required by the application e.g. writing a database record or generating and submitting a command.
+  #. Make a small change to your DAML data model
+  #. Optionally add a scenario unit test to assert correctness
+  #. Adapt the UI such that the code compiles again against the new DAML code
+  #. Extend the UI to make use of the newly introduced feature
+  #. Restart with step 1
```

Comment:
```suggestion
  #. Make further changes either to your DAML and/or React code until you're happy with what you've developed
```
Just wanna write something so that it doesn't sound like users need to do something, this step is optional 😄